hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java

```diff
@@ -551,6 +556,24 @@ public IOStatistics getIOStatistics() {
     return iostatistics;
   }
 
+  @Override
+  public void abort() {
```

Comment:
Migrating comment in https://github.com/HeartSaVioR/hadoop/commit/63c5588d46d422a66ec620ae310f9a860db890b3#r46507150

> the new "cloud ready" API calls always return a CompletableFuture, to emphasise that the op may take time and to allow the caller to do something while waiting. Would we want to do this here? I'm not convinced it is appropriate. Instead we say
>
> 1. call must guarantee that after this is invoked,. close() will not materialize the file at its final path
> 2. it may communicate with the store to cancel an operation; which may retry. Errors will be stored.
> 3. there may still/also be async IO to the store after the call returns, but this must maintain the requirement "not visible"
> 4. And close() may do some IO to cancel